### PR TITLE
OKTA-00 syncer for integrating with Okta groups via api token

### PR DESCRIPTION
**Pull request by vianuevm**
Hi all :) 

This commit introduces a syncer that syncs Okta groups to a specific application into OpenShift.  For example, if you used Okta as your SSO for OCP, then it would sync the groups associated with that Okta OCP application into OpenShift.

This workflow uses an Okta API Token.

### CODE REVIEW

**Comment by Jmainguy** [comment]
We are currently using a version of this at Bandwidth to Okta groups pulled in, and seems to be working pretty good so far =). Nice work Mike. We hope to get this merged so we can use upstream and not rely on a fork going forward.

File: README.md [hunk 1]
```diff
@@ -389,6 +390,52 @@ The secret can be created by executing the following command:
...
+### Okta
+
+Groups assigned to Okta applications can be synchronized into OpenShift. The following table describes the set of configuration options for the Okta provider:
+
+| Name | Description | Defaults | Required | 
+| ----- | ---------- | -------- | ----- |
+| `credentialsSecret` | Reference to a secret containing authentication details (See below) | | Yes |
+| `groups` | List of groups to filter against | | No |
+| `scope` | Scope for group synchronization. Options are `one` for one level or `sub` to include subgroups | `sub` | No |
+| `url` | URL Location for Okta | | Yes |
```
Comment (sabre1041) [inline README.md:402]: Need to be explicit on the format that is required (scheme and trailing `/`)

File: README.md [hunk 2]
```diff
@@ -389,6 +390,52 @@ The secret can be created by executing the following command:
 oc create secret generic keycloak-group-sync --from-literal=username=<username> --from-literal=password=<password>
 ```
 
+### Okta
+
+Groups assigned to Okta applications can be synchronized into OpenShift. The following table describes the set of configuration options for the Okta provider:
+
+| Name | Description | Defaults | Required |
```
Comment (sabre1041) [inline README.md:397]: Be sure to include all default values

File: README.md [hunk 3]
```diff
@@ -389,6 +390,52 @@ The secret can be created by executing the following command:
...
+metadata:
+  name: okta-sync
+spec:
+  providers:
+    - name: okta
+      okta:
+        credentialsSecret:
+          name: okta-api-token
+          namespace: group-sync-operator
+        url: "example.okta.com"
```
Comment (sabre1041) [inline README.md:423]: Specifying this format throws errors due to invalid URI

File: README.md [hunk 4]
```diff
@@ -389,6 +390,52 @@ The secret can be created by executing the following command:
 oc create secret generic keycloak-group-sync --from-literal=username=<username> --from-literal=password=<password>
 ```
 
+### Okta
+
+Groups assigned to Okta applications can be synchronized into OpenShift. The following table describes the set of configuration options for the Okta provider:
```
Comment (sabre1041) [inline README.md:395]: Any links provided by Okta that can be shared with end users?

**Review by sabre1041 — CHANGES_REQUESTED** [review]
This looks really good. A few comments that require addressing. 

Would you also align the documentation to the values provided in the [ClusterServiceVersion](https://github.com/redhat-cop/group-sync-operator/blob/master/config/manifests/bases/group-sync-operator.clusterserviceversion.yaml)

File: config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml [hunk 5]
```diff
@@ -700,6 +700,54 @@ spec:
                   name:
                     description: Name represents the name of the provider
                     type: string
+                  okta:
```
Comment (vianuevm) [inline config/crd/bases/redhatcop.redhat.io_groupsyncs.yaml:703]: Removed the unused variable scope.

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 6]
```diff
@@ -347,6 +347,31 @@ func (in *LdapProvider) DeepCopy() *LdapProvider {
 	return out
```
Comment (vianuevm) [inline api/v1alpha1/zz_generated.deepcopy.go:347]: File regenerated due to removing unused variable scope.

File: README.md (hunk 2 above)
Comment (vianuevm) [reply README.md:398]: One field has a default value that I just added, the others use Go's zero values.  Let me know if you'd like me to specify that (false for extractLoginUsername, for example).

File: README.md (hunk 1 above)
Comment (vianuevm) [reply README.md]: Updated this in the examples and added trailing slash requirement

File: README.md (hunk 4 above)
Comment (vianuevm) [reply README.md]: Added some additional links that I think will be help around groups/apps, as we well as docs for the API that `okta.go` calls out to

File: README.md (hunk 2 above)
Comment (sabre1041) [reply README.md:398]: If you could specify `false` (it doesnt have to be explicit), but should at least be noted in the README

File: config/samples/okta.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  providers:
+    - name: okta
+      okta:
+        credentialsSecret:
+          name: okta-api-token
+          namespace: group-sync-operator
+        url: "https://example.okta.com/"
+        appId: okta-sync-app-id
+        emailUsernameSplit: true
```
Comment (sabre1041) [inline config/samples/okta.yaml:14]: This should be `extractLoginUsername`

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Missed this one. This change and then the other README change and we should be good to go

File: README.md (hunk 2 above)
Comment (vianuevm) [reply README.md:398]: Updated all the fields to have the default value in the table 👍

File: config/samples/okta.yaml (hunk 7 above)
Comment (vianuevm) [reply config/samples/okta.yaml]: Good catch!  Fixed 👍

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**